cmd/queue/main.go

```diff
@@ -243,39 +226,61 @@ func handler(reqChan chan queue.ReqEvent, breaker *queue.Breaker, handler http.H
 }
 
 // Sets up /health and /wait-for-drain endpoints.
-func createAdminHandlers() *http.ServeMux {
+func createAdminHandlers(p *readiness.Probe) *http.ServeMux {
 	mux := http.NewServeMux()
-	// TODO(@joshrider): temporary change while waiting on other PRs to merge (See #4014)
-	mux.HandleFunc(requestQueueHealthPath, healthState.HealthHandler(probeUserContainer, true /*isNotAggressive*/))
+
+	mux.HandleFunc(requestQueueHealthPath, healthState.HealthHandler(p.ProbeContainer, p.IsAggressive()))
 	mux.HandleFunc(queue.RequestQueueDrainPath, healthState.DrainHandler())
 
 	return mux
 }
 
-func probeQueueHealthPath(port int, timeout time.Duration) error {
+func probeQueueHealthPath(port int, timeoutSeconds int) error {
 	url := fmt.Sprintf(healthURLTemplate, port)
 
+	// Use aggressive sub-second retries.
+	if timeoutSeconds == 0 {
```

Comment:
Hmm, why do we need two paths here? The HTTP request to `/health` should be retriable as quickly as we want, no? At least we should be able to retry it as quickly as we want before it responds at all. Once it starts responding false, we can pick between the fast path (agressive retries) or the defined period.

Even better: Now that we have in-path probes for the activator for example, should we use that path here? It'd also make sure the "main" HTTP server is actually up and running.

In general I'd seek for unification between the existing "aggressive" probing and the new things. In the best case, our own probe behaves the same as a user-set TCP probe with periodSeconds = 0.